### ISSUE TITLE
The DBUnit "Table Type" property is not supported

### ISSUE THREAD

**Issue by mats24** [database-rider/database-rider#235]
I need to change the DBUnit table type from the default {"TABLE"} to {"TABLE", "VIEW"}, as the database "table" I'm manipulating is actually an Oracle view: http://dbunit.sourceforge.net/properties.html#tabletype

However, the database-rider DBUnit annotation does not include a "tableType" attribute.
Is this going to be added in a future release? Or is there some other way of configuring the missing DBUnit properties?

Environment: Spring Boot, JUnit 5, DBUnit, Database Rider.

**Comment by rmpestano** [comment]
Hi @mats24, you can add dbunit properties in [global config file](https://github.com/database-rider/database-rider#332-dbunit-configuration) (`src/test/resources/dbunit.yml`), for example:

```
cacheConnection: false
cacheTableNames: false
leakHunter: true
raiseExceptionOnCleanUp: true
expectedDbType: HSQLDB
properties:
  tableType: {"TABLE", "VIEW"}
  batchedStatements:  true
  caseSensitiveTableNames: false
  qualifiedTableNames: true
  schema: public
  batchSize: 200
  fetchSize: 200
  allowEmptyFields: true
  escapePattern: "[?]"
  datatypeFactory: !!com.github.database.rider.core.configuration.DBUnitConfigTest$MockDataTypeFactory {}
  replacers: [!!com.github.database.rider.core.replacers.CustomReplacer {}]
```

**Comment by mats24** [comment]
Thanks - that link doesn't explicitly mention the tableType property so I assumed it wasn't supported in the yml, either.

I've been trying what you suggested; however, I can't get `tableType `to be read as a String array, which is what's expected.

When I specify as: 
```
properties:
  tableType: {"TABLE", "VIEW"}

```
I get:
```
java.lang.IllegalArgumentException: Cannot cast object of type 'class java.util.LinkedHashMap' to allowed type 'class [Ljava.lang.String;'.

	at org.dbunit.database.DatabaseConfig.checkObjectAllowed(DatabaseConfig.java:306)
	at org.dbunit.database.DatabaseConfig.setProperty(DatabaseConfig.java:233)
	at com.github.database.rider.core.connection.RiderDataSource.configDatabaseProperties(RiderDataSource.java:95)
	at com.github.database.rider.core.connection.RiderDataSource.initDBUnitConnection(RiderDataSource.java:84)
	at com.github.database.rider.core.connection.RiderDataSource.init(RiderDataSource.java:77)
	at com.github.database.rider.core.connection.RiderDataSource.<init>(RiderDataSource.java:45)
	at com.github.database.rider.core.dataset.DataSetExecutorImpl.getRiderDataSource(DataSetExecutorImpl.java:1064)
	at com.github.database.rider.core.RiderRunner.setup(RiderRunner.java:31)
	at com.github.database.rider.junit5.DBUnitExtension.beforeTestExecution(DBUnitExtension.java:64)

```
etc.


If I try:
```
properties:
  tableType: ["TABLE", "VIEW"]

```
I get:
`java.lang.IllegalArgumentException: Cannot cast object of type 'class java.util.ArrayList' to allowed type 'class [Ljava.lang.String;'.`




If I try:
```
properties:
  tableType:
    - "TABLE"
    - "VIEW"

```
I get:
`java.lang.IllegalArgumentException: Cannot cast object of type 'class java.util.ArrayList' to allowed type 'class [Ljava.lang.String;'.`



Could you please suggest how tableType can be successfully formatted?

Thanks

**Comment by rmpestano** [comment]
Hi @mats24, can you try with `1.16.1-SNAPSHOT`? You'll need to enable snapshots repository on your build:

```
<repositories>
    <repository>
        <snapshots/>
        <id>snapshots</id>
        <name>libs-snapshot</name>
        <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    </repository>
</repositories>
```

Now the following should work:

```
properties:
  tableType: ["TABLE", "VIEW"]
```

Also it's now possible to configure tableType via `@DBUnit` annotation as well.

**Comment by mats24** [comment]
Thanks for the update - that sounds encouraging.
Unfortunately I can't access the [https://oss.sonatype.org/content/repositories/snapshots] repository, so my project cannot resolve the rider-core 1.16.1-SNAPSHOT dependency.
Could you please help me with how I can access that?
Thanks

**Comment by rmpestano** [comment]
Did you added snapshots repository on your pom.xml?

```
<repositories>
    <repository>
        <snapshots/>
        <id>snapshots</id>
        <name>libs-snapshot</name>
        <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    </repository>
</repositories>
```

**Comment by mats24** [comment]
It turns out my Maven mirror was causing the issue; I've now configured it to exclude the above `snapshots` repository and it now resolves the new rider-core snapshot successfully.
I've retested `tableType` with both dbunit.yml and the `@DBUnit` annotation, and can confirm that both now work successfully.
Thanks for your help.

**Comment by rmpestano** [comment]
Nice! There will be a new release with this feature soon. Thank you for the feedback!

**Comment by rmpestano** [comment]
This issue is fixed in `v1.16.1` which was just released, thank you @mats24!
